_projects/git_task_list.md

```diff
@@ -0,0 +1,82 @@
+difficulty: medium
+initiatives:
+  - GSoC
+issues:
+  - https://github.com/coala/projects/issues/480
+  - https://github.com/coala/coala/issues/3615
+  - https://github.com/coala/coala/issues/4080
+markdown: git_task_list.md
+mentors:
+  - sims1253
+  - yukiisbored
+  - yashovardhanagrawal
+name: "Git Task List"
+requirements:
+  - "The applicant should have at least one patch merged to any of the coala repositories."
+  - "Familiarity with Git, GitHub API Gerrit API, Phabricator API, and GitLab API"
+  - "Familiarity with EmberJS, Javascript, and GitHub Pages"
+status: in_progress
+tags:
+  - community
+  - web
+---
+The basic idea is an extended version of [GSOC Task List](http://summerofcode.github.io/gsoc-prep-tasks)
+which was initialy intended for GSOC Applicans who don't have or only have a
+litle experience in open-source. Hence, [that](http://github.com/summerofcode/gsoc-prep-tasks)
+project will be the starting point of this project.
+
+This project will help sharing the tasks of reviewing a pull request on a
+project. It will help fix the problem where developer spend too much time
+finding PR that haven’t been reviewed. This projects aims to getting more
+developer doing reviews works and sharing reviews workload, by providing an easy
+way to find issues that has not been reviewed. It also help project’s maintainer
+/ developer do review more efficient.
+
+This also help engage developer to keep contributing to a project where they
+can’t find any issue that they can fix or keep the new comer issue for new comer
+as learning path. And developer with more experience can contribute by reviewing
+new comer’s work PR
+
+This project is also related to “New Comer Metrics And Gamification”. Where some
+of the activities that will be given some point is involving:
+
+- Issues
+  - Comment to an Issue
+  - Create new Issue
+- Pull request
+  - Review to a Pull Request
+  - Create a Pull request
+
+Which is addressed by this project.
+
+The Goal of this projects is a serverless web page where the user can find Pull
+Request that haven’t been reviewed. User can select projects in Open Source
+hosting services like GitHub or GitLab. They also have the ability to select
+prefered programming language, tags, owner of the pull requests, and other
+metadata that is provided by the web services (Github/Gitlab).
+
+#### Milestone
+
+##### Phase 1
+- Add GitLab support
+  - Search Projects
+  - Search open issues
+  - Search issue by tag
+- Add complete list of GSOC organizations that is using GitHub or GitLab
+  - Find GitLab / GitHub repository pages
+  - Add simplified ui to list/search projects
+
+##### Phase 2
+- PR review list / queue
+- Create a new search page for PR
+- Search unreviewed PR
+- Search unreviewed PR on a specific projects
+
+##### Phase 3
+- Add Gerrit support
+- Add Phabricator support
```

Comment:
Is this required?

Reply:
In my opinion, Not required as the usage in coala is small. GSOC-prep-task was outside of coala. I did added it to my GSOC proposal.
Can be raplaced to something else that is more integrated with coala

Reply:
coala builds tools for other orgs, and that is what your project had planned to do.
Ideally this tool supports 'task' system of many GSoC organisations.  Then the tool can be a centralised search tool for potential GSoC students to find tasks in lots of organisations, helping them find the best organisation that interests them.

GitHub Issues is obviously the most common ; GitLab is important ; a few use Phabricator, and it is becoming more popular.  In addition to Wikimedia, the BSDs projects are using it.  I know you can use the GSOC json data to identify orgs which use Phabricator.

ALso note that Phabricator has both tasks and git hosting and code review.